### ISSUE TITLE
Qgis crash when deleting columns in a temporary layer

### ISSUE THREAD

**Issue by arer01** [qgis/QGIS#56881]
### What is the bug or the crash?

The program crash when deleting all but one column in a temporary layer

### Steps to reproduce the issue

Create a buffer
Cut another layer by the buffer
Edit the new layer and when you delete all but one column the program crashes

### Versions

QGIS Version: 3.28.15-Firenze

### Supported QGIS version

- [ ] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [ ] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

## User Feedback

Deleting all but one column in a temporary layer 

## Report Details

**Python Stack Trace**
```
Windows fatal exception: access violation

Thread 0x00001390 (most recent call first):
<no Python frame>
Windows fatal exception: access violation

Thread 0x00001390 (most recent call first):
<no Python frame>
Windows fatal exception: code 0xc0000374

Current thread 0x00001390 (most recent call first):
<no Python frame>
```

**Stack Trace**

```

sqlite3_bind_int64 :
sqlite3_bind_int64 :
sqlite3_bind_int64 :
sqlite3_bind_int64 :
sqlite3_bind_int64 :
sqlite3_bind_int64 :
sqlite3_bind_int64 :
CPLODBCDriverInstaller::RemoveDriver :
CPLODBCDriverInstaller::RemoveDriver :
QgsSQLiteExpressionCompiler::operator= :
QgsAbstractFeatureIterator::nextFeature :
QgsVectorLayerFeatureIterator::fetchFeature :
QgsAbstractFeatureIterator::nextFeature :
QgsVectorLayerProfileResults::visitFeaturesInRange :
QgsVectorLayerProfileResults::visitFeaturesInRange :
QgsVectorLayerProfileResults::visitFeaturesInRange :
QgsMapRendererParallelJob::renderLayerStatic :
QgsProfilePlotRenderer::replaceSourceInternal :
QgsMapRendererParallelJob::renderingFinished :
QgsMapRendererParallelJob::takeLabelingResults :
QtConcurrent::ThreadEngineBase::run :
QThreadPoolPrivate::reset :
QThread::start :
BaseThreadInitThunk :
RtlUserThreadStart :
```

**QGIS Info**
QGIS Version: 3.28.15-Firenze
QGIS code revision: 252ad49ddc
Compiled against Qt: 5.15.3
Running against Qt: 5.15.3
Compiled against GDAL: 3.8.3
Running against GDAL: 3.8.3

**System Info**
CPU Type: x86_64
Kernel Type: winnt
Kernel Version: 10.0.19045

**Comment by agiudiceandrea** [comment]
@arer01, thanks for reporting. QGIS 3.28 is no longer supported. Please try if the issue occurs running a currently supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule) and using a new [QGIS user profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles). If it does, please provide the layers and exact steps and parameters with which the issue occurs.

**Comment by arer01** [comment]
Hello thanks for reminding me, the issue solved it self when updating to 3.34.4
